pkg/trait/trait_types.go

```diff
@@ -95,40 +95,49 @@ type Trait interface {
 
 	// RequiresIntegrationPlatform indicates that the trait cannot work without an integration platform set
 	RequiresIntegrationPlatform() bool
+
+	// IsAllowedInProfile tels if the trait supports the given profile
+	IsAllowedInProfile(v1.TraitProfile) bool
+
+	// Order is the order in which the trait should be executed in the normal flow
+	Order() int
 }
 
 /* Base trait */
 
-func newBaseTrait(id string) BaseTrait {
+// NewBaseTrait --
+func NewBaseTrait(id string, order int) BaseTrait {
```

Comment:
Wonder if we should have some constants that could help choosing the right value for the `order` field

Reply:
I was thinking the same, but couldn't find a good strategy since we have so many traits ;)

Reply:
I've opted for defining some relevant constants that addons may need (I've though to the cases mentioned above), but still use numbers for the core. Have a look.

Reply:
LGTM, so as future step we could think to generalize this pattern as is exactly the same we are using for the builder so we should probably come up with a sort of a finite step machine